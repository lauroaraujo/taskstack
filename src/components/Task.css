.PENDING {
    color: darkcyan;
}

.BLOCKED {
    color: blueviolet;
    opacity: 0.5; 
}

.DONE {
    text-decoration: line-through;
    opacity: 0.5;
    color: green;
}

.DROPPED {
    color: red;
    text-decoration: line-through;
    opacity: 0.5;
}