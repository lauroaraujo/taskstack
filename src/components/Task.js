import React from 'react';
import PropTypes from 'prop-types';
import './Task.css';

export const TASK_STATUS = {
  PENDING: 'PENDING',
  BLOCKED: 'BLOCKED',
  DONE: 'DONE',
  DROPPED: 'DROPPED',
};

export const TASK_STATUS_LIST = Object.keys(TASK_STATUS);

const Task = ({ content, status }) => (
  <div className={status}>
    {content}
  </div>
);

Task.propTypes = {
  content: PropTypes.string,
  status: PropTypes.oneOf(TASK_STATUS_LIST),
};

export default Task;
