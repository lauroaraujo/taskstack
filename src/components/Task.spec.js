import React from 'react';
import { shallow } from 'enzyme';
import { Task, TASK_STATUS } from './Task';

describe('Task', () => {
  it('renders pending task', () => {
    const task = shallow(<Task id={1} status={TASK_STATUS.PENDING} content='I should do this soon.' />);
    expect(task).toMatchSnapshot();
  });

  it('renders blocked task', () => {
    const task = shallow(<Task id={2} status={TASK_STATUS.BLOCKED} content={'I will do this after that.'} />);
    expect(task).toMatchSnapshot();
  });

  it('renders done task', () => {
    const task = shallow(<Task id={3} status={TASK_STATUS.DONE} content='I did this a while back.' />);
    expect(task).toMatchSnapshot();
  });

  it('renders dropped task', () => {
    const task = shallow(<Task id={4} status={TASK_STATUS.DROPPED} content={'I don\'t need to do this anymore.'} />);
    expect(task).toMatchSnapshot();
  });

  it('toggles status on input change', () => {
    const setStatus = jest.fn();
    const id = 11;
    const task = shallow(<Task
      id={id}
      status={TASK_STATUS.PENDING}
      setStatus={setStatus}
    />);

    const taskInput = task.find('input');
    const event = {
      stopPropagation: jest.fn(),
    }

    taskInput.simulate('change', event);
    expect(setStatus).toHaveBeenLastCalledWith(id, TASK_STATUS.BLOCKED);
    expect(event.stopPropagation).toHaveBeenCalledTimes(1);

    task.setProps({status: TASK_STATUS.BLOCKED});
    taskInput.simulate('change', event);
    expect(setStatus).toHaveBeenLastCalledWith(id, TASK_STATUS.DONE);
    expect(event.stopPropagation).toHaveBeenCalledTimes(2);

    task.setProps({status: TASK_STATUS.DONE});
    taskInput.simulate('change', event);
    expect(setStatus).toHaveBeenLastCalledWith(id, TASK_STATUS.DROPPED);
    expect(event.stopPropagation).toHaveBeenCalledTimes(3);

    task.setProps({status: TASK_STATUS.DROPPED});
    taskInput.simulate('change', event);
    expect(setStatus).toHaveBeenLastCalledWith(id, TASK_STATUS.PENDING);
    expect(event.stopPropagation).toHaveBeenCalledTimes(4);

    task.setProps({status: undefined});
    taskInput.simulate('change', event);
    expect(setStatus).toHaveBeenLastCalledWith(id, undefined);
    expect(event.stopPropagation).toHaveBeenCalledTimes(5);
  });
});
