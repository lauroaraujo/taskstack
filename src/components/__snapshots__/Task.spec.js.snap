// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Task renders blocked task 1`] = `
<div
  className="BLOCKED"
>
  I will do this after that.
</div>
`;

exports[`Task renders done task 1`] = `
<div
  className="DONE"
>
  I did this a while back.
</div>
`;

exports[`Task renders dropped task 1`] = `
<div
  className="DROPPED"
>
  I don't need to do this anymore.
</div>
`;

exports[`Task renders pending task 1`] = `
<div
  className="PENDING"
>
  I should do this soon.
</div>
`;
